src/basic/virt.c

```diff
@@ -472,6 +472,20 @@ int detect_container(void) {
                 goto finish;
         }
 
+        /* Docker creates /.dockerenv or /run/.dockerenv inside the container.
+         * https://github.com/moby/moby/issues/18355 */
+        if (access("/.dockerenv", F_OK) >= 0 || access("/run/.dockerenv", F_OK) >= 0) {
```

Comment:
Yuck, it's a pity the docker people are unwilling to play nice and just set `$container` like everyone else.

Reply:
+1